retrieval/discovery/kubernetes/endpoints.go

```diff
@@ -83,23 +83,50 @@ func (e *Endpoints) Run(ctx context.Context, ch chan<- []*config.TargetGroup) {
 
 	e.endpointsInf.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(o interface{}) {
-			send(e.buildEndpoints(o.(*apiv1.Endpoints)))
+			eps, err := convertToEndpoints(o)
+			if err != nil {
+				e.logger.With("err", err).Errorln("converting to Endpoints object failed")
+				return
+			}
+			send(e.buildEndpoints(eps))
 		},
 		UpdateFunc: func(_, o interface{}) {
-			send(e.buildEndpoints(o.(*apiv1.Endpoints)))
+			eps, err := convertToEndpoints(o)
+			if err != nil {
+				e.logger.With("err", err).Errorln("converting to Endpoints object failed")
+				return
+			}
+			send(e.buildEndpoints(eps))
 		},
 		DeleteFunc: func(o interface{}) {
-			send(&config.TargetGroup{Source: endpointsSource(o.(*apiv1.Endpoints).ObjectMeta)})
+			eps, err := convertToEndpoints(o)
+			if err != nil {
+				e.logger.With("err", err).Errorln("converting to Endpoints object failed")
+				return
+			}
+			send(&config.TargetGroup{Source: endpointsSource(eps)})
 		},
 	})
 
-	serviceUpdate := func(svc *apiv1.Service) {
+	serviceUpdate := func(o interface{}) {
+		svc, err := convertToService(o)
+		if err != nil {
+			e.logger.With("err", err).Errorln("converting to Service object failed")
+			return
+		}
+
 		ep := &apiv1.Endpoints{}
 		ep.Namespace = svc.Namespace
 		ep.Name = svc.Name
 		obj, exists, err := e.endpointsStore.Get(ep)
 		if exists && err != nil {
-			send(e.buildEndpoints(obj.(*apiv1.Endpoints)))
+			eps, err := convertToEndpoints(obj)
```

Comment:
Removed this as discussed offline with @fabxc , because we are relying in many places that at least the store stores consistently typed objects.